framework/assets/yii.js

```diff
@@ -232,49 +230,41 @@ window.yii = (function ($) {
 
             var activeFormData = $form.data('yiiActiveForm');
             if (activeFormData) {
-                // remember who triggers the form submission. This is used by yii.activeForm.js
+                // Remember the element triggered the form submission. This is used by yii.activeForm.js.
                 activeFormData.submitObject = $e;
             }
 
-            // temporarily add hidden inputs according to data-params
-            if (params && $.isPlainObject(params)) {
-                $.each(params, function (idx, obj) {
-                    $form.append($('<input/>').attr({name: idx, value: obj, type: 'hidden'}));
+            if (areValidParams) {
+                $.each(params, function (name, value) {
+                    $form.append($('<input/>').attr({name: name, value: value, type: 'hidden'}));
                 });
             }
 
-            var oldMethod = $form.attr('method');
-            $form.attr('method', method);
-            var oldAction = null;
-            if (action && action != '#') {
-                oldAction = $form.attr('action');
-                $form.attr('action', action);
-            }
             if (usePjax) {
-                $form.on('submit',function(e){
+                $form.on('submit', function (e) {
                     $.pjax.submit(e, pjaxOptions);
-                })
+                });
             }
+
             $form.trigger('submit');
-            $.when($form.data('yiiSubmitFinalizePromise')).then(
-                function () {
-                    if (oldAction != null) {
-                        $form.attr('action', oldAction);
-                    }
-                    $form.attr('method', oldMethod);
 
-                    // remove the temporarily added hidden inputs
-                    if (params && $.isPlainObject(params)) {
-                        $.each(params, function (idx, obj) {
```

Comment:
`idx` and `obj` are not appropriate for params stored in `name` - `value` structure.